external_search.py

```diff
@@ -640,21 +646,16 @@ def _get_raw_results():
         def _count_docs():
             # The mock methods used in testing return a list, so we have to call len() rather than count().
             if in_testing:
-                return str(len(_works()))
-
-            return str(_works().count())
+                return str(len(self.search))
+            return str(self.search.count())
 
         yield self.run_test(
             ("Total number of search results for '%s':" %(self.test_search_term)),
             _count_docs
         )
 
         def _total_count():
-            # The mock methods used in testing return a list, so we have to call len() rather than count().
-            if in_testing:
-                return str(len(self.search))
-
-            return str(self.search.count())
+            return str(self.count_works(None))
```

Comment:
Here we don't need to treat the mocked method differently, because there's now a special method to count all the works in the index, and we mock that method.